### ISSUE TITLE
unclean reactor error from test_asynchronousException on OS X

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2869]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2869|
|Type|defect|
|Created|2007-11-05 02:35:47Z|
|Branch|https://github.com/twisted/twisted/tree/pbfailure-unclean-2869|

===============================================================================
[ERROR]: twisted.test.test_pbfailure.PBFailureTestUnsafe.test_asynchronousException

Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
Selectables:
<Broker #0 on 0>
-------------------------------------------------------------------------------

<details><summary>Searchable metadata</summary>

```
trac-id__2869 2869
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_pbfailure_unclean_2869 branches/pbfailure-unclean-2869
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1194230147000000 1194230147000000
changetime__1210083052000000 1210083052000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [23445]) Branching to 'pbfailure-unclean-2869'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [23446]) Keep track of the connection server-side and clean it up in tearDown

Refs [#2869](https://github.com/twisted/twisted/issues/2869)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve removed owner|
|-|-|

All the tests in pbfailure could failed at some point. I think I fixes it in the branch, please review!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

[#2917](https://github.com/twisted/twisted/issues/2917) has been closed as a duplicate of this one.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @therve|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set status to `closed`|
|-|-|

(In [23526]) Merge pbfailure-unclean-2869

Author: therve
Reviewer: exarkun
Fixes [#2869](https://github.com/twisted/twisted/issues/2869)

Remove some intermittent unclean errors in twisted.test.test_pbfailure where
a server-side connection was left around.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
